### PR TITLE
Normalize ilvaccine org for IL

### LINKED ISSUE #315: normalize il/ilvaccine_org
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #314

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch il/ilvaccine_org
    poetry run vaccine-feed-ingest parse il/ilvaccine_org
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize il/ilvaccine_org
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by cdjj**
| Key Details |
|-|
| Resolves #315  |
State: IL |
Site: ilvaccine_org |

## Notes
The input data is a bit messy on addresses, so the regex unfortunately misses about half a dozen records.  It would be nice if we had a standard normalizer (e.g. calling `usaddress`) that we could just feed these into and get reasonable output rather than messy regexen.  Ah well.

## Data sample
```json
{"id": "ilvaccine_org:rec5FTKvH6e2hoUZm", "name": "Advocate Condell Medical Center COVID-19 Vaccine Center", "address": {"street1": "700 Garfield Ave", "street2": null, "city": "Libertyville", "state": "IL", "zip": "60048"}, "location": null, "contact": [{"contact_type": null, "phone": null, "website": "https://www.advocateaurorahealth.org/coronavirus-disease-2019/covid19-vaccine", "email": null, "other": null}], "languages": null, "opening_dates": null, "opening_hours": null, "availability": null, "inventory": null, "access": null, "parent_organization": null, "links": null, "notes": ["Advocate Aurora Health is now vaccinating anyone in Illinois who\u2019s 16 and older \u2013 whether you\u2019re their patient or not."], "active": null, "source": {"source": "ilvaccine_org", "id": "rec5FTKvH6e2hoUZm", "fetched_from_uri": "https://ilvaccine-api.us-east-1.linodeobjects.com/vts.ndjson", "fetched_at": "2021-05-25T08:36:49.142761", "published_at": null, "data": {"id": "ilvaccine_org:rec5FTKvH6e2hoUZm", "name": "Advocate Condell Medical Center COVID-19 Vaccine Center", "address": "700 Garfield Ave\nLibertyville, IL 60048", "contact": [{"website": "https://www.advocateaurorahealth.org/coronavirus-disease-2019/covid19-vaccine"}], "notes": ["Advocate Aurora Health is now vaccinating anyone in Illinois who\u2019s 16 and older \u2013 whether you\u2019re their patient or not."], "source": {"source": "ilvaccine_org", "id": "rec5FTKvH6e2hoUZm"}}}}
```

## Before Opening a PR
- [X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by eliblock** [comment]
@cdjj we added `usaddress` in #600, and the code for using it looks relatively straightforward.

Are you interested in trying to use it here? Once it's been used in a couple of places we can try to extract to a shared normalizer util as you suggest.

If not, I'll review this PR!

**Comment by cdjj** [comment]
Let's do this first, then I'll work on extracting a shared normalizer and do a cleanup pass, like we did for phone numbers.

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ADDRESS_RE = re.compile(r"(.*)(?:,|\n)([a-zA-Z. ]+),\s?(?:IL|Illinois)?\s+(\d{5})")
+
+# Rewrites for addresses that don't quite fit the regex.
+ADDRESS_FIXES = {
+    "3401 N Knox Ave Chicago, IL 60641": "3401 N Knox Ave\nChicago, IL 60641",
+    "2674 N Halsted St Chicago IL 60614\n": "2674 N Halsted St\nChicago, IL 60614",
+    "701 W North Ave, Melrose Park, IL": "701 W North Ave, Melrose Park, IL 60160",
+    ".": "2751 W Winona St, Chicago, IL 60625",  # Swedish Hospital, Chicago
+    "2000 5th Ave River Grove, IL 60171": "2000 5th Ave\nRiver Grove, IL 60171",
+    "3128 Voyager Lane\nJoliet, IL": "3128 Voyager Lane\nJoliet, IL 60431",
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:26]: @edcous - could these six addresses be fixed upstream in the data source?

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+logger = getLogger(__file__)
+
+ADDRESS_RE = re.compile(r"(.*)(?:,|\n)([a-zA-Z. ]+),\s?(?:IL|Illinois)?\s+(\d{5})")
+
+# Rewrites for addresses that don't quite fit the regex.
+ADDRESS_FIXES = {
+    "3401 N Knox Ave Chicago, IL 60641": "3401 N Knox Ave\nChicago, IL 60641",
+    "2674 N Halsted St Chicago IL 60614\n": "2674 N Halsted St\nChicago, IL 60614",
+    "701 W North Ave, Melrose Park, IL": "701 W North Ave, Melrose Park, IL 60160",
+    ".": "2751 W Winona St, Chicago, IL 60625",  # Swedish Hospital, Chicago
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:24]: I'm a little nervous about mapping `.` to something incase the upstream issue re-occurs. Would it maybe be better to drop this particular address?

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    if site.get("notes"):
+        return site["notes"]
+    return None
+
+
+def _get_source_id(site: dict) -> str:
+    if site.get("source"):
+        return site["source"]["id"]
+    elif site.get("source1"):
+        return site["source1"]["id"]
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:40]: @edcous - why does the data sometimes have `source` and sometimes have `source1`? Is this intentional?

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+
+def _get_notes(site: dict) -> Optional[List[str]]:
+    if site.get("notes"):
+        return site["notes"]
+    return None
+
+
+def _get_source_id(site: dict) -> str:
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:36]: Should we pull this from the second half of the `id` instead?

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import sys
+from typing import List, Optional
+
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+ADDRESS_RE = re.compile(r"(.*)(?:,|\n)([a-zA-Z. ]+),\s?(?:IL|Illinois)?\s+(\d{5})")
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:17]: Nice regex!

**Review by eliblock — CHANGES_REQUESTED** [review]
Overall looks good, thank you for writing this! I'm hesitant to merge with logic that re-maps `.` to a single address, but if you're okay with removing that, we can get this merged!

Additionally, I tagged @edcous with a couple questions. They maintain the upstream data source, so may be able to help us out here!

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py (hunk 4 above)
Comment (cdjj) [reply vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py:35]: SGTM, done.

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py (hunk 3 above)
Comment (edcous) [reply vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py]: Sorry about that, fixed.

File: vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py (hunk 1 above)
Comment (edcous) [reply vaccine_feed_ingest/runners/il/ilvaccine_org/normalize.py]: Yup, I can do that now.

**Review by edcous — APPROVED** [review]
Thanks for doing this!

**Review by eliblock — APPROVED** [review]
Thank you!

I suspect we can clean up some of the resiliency to upstream quirks now that they've been cleaned up... but no harm in leaving them in for now. 

Let's get these sites on the map!

**Pull request merged**